src/wallet/rpcdump.cpp

```diff
@@ -114,6 +115,14 @@ UniValue importprivkey(const UniValue& params, bool fHelp)
     if (params.size() > 2)
         fRescan = params[2].get_bool();
 
+    CBlockIndex *pindex = chainActive.Genesis();
+    if (params.size() > 3) {
+        int nTime = params[3].get_int();
+        pindex = chainActive.FindLatestBefore(nTime);
+        if (!pindex)
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "No block before timestamp");
```

Comment:
It's also a useless error - the user doesn't care if there was a block before their timestamp... just rescan the entire chain in that case.